Added all upgrade docs (#4925)
umbraco/UmbracoDocs: 10/umbraco-cms/fundamentals/setup/upgrading/version-specific/minor-upgrades-for-umbraco-7.md

## Contents

In this article you will find instruction of 2 different ways of upgrading:

* [Upgrade using NuGet](#upgrade-using-nuget)
* [Upgrade manually from a Zip file](#upgrade-manually-from-a-zip-file)

## Upgrade using NuGet

1. Open up the **Package Console** and type: `Update-Package UmbracoCms`
2. Choose **"No to All"** by pressing the **"L"** when prompted.
    * If there are any specific configuration changes required for the version you are upgrading to then they will be noted in the **[version-specific guide](version-specific.md)**.

Alternatively, you can use the Visual Studio **NuGet Package Manager** to upgrade:

1. Open the **NuGet Package Manager** and select the **Updates** pane to get a list of available updates.
2. Choose the package called **UmbracoCms** and select update.

The upgrade will run through all the files and make sure you have the latest changes while leaving files you have updated.

### Upgrades to versions lower than 7.2.0

> If you're not upgrading to 7.2.0 or higher then you should follow these extra instructions, if you are upgrading to 7.2.0+ then you can skip this and go to [Merge UI.xml and language](#merge-ui-xml-and-language).

Reviewer: ```suggestion
If you're not upgrading to 7.2.0 or higher then you should follow these extra instructions. If you are upgrading to 7.2.0+ then you can skip this and go to [Merge UI.xml and language](#merge-ui-xml-and-language).
```

> **The following only applies to upgrading to versions lower than 7.2.0**
> You will be asked to overwrite your web.config file and the files in /config, make sure to answer **No** to those questions.

Reviewer: ```suggestion
### The following only applies to upgrading to versions lower than 7.2.0 

You will be asked to overwrite your web.config file and the files in /config, make sure to answer **No** to those questions.
```

For some inexplicable reason, the installation will fail if you click "No to All" (in the GUI) or answer "L" (in the package manager console) to the question: "File 'Web.config' already exists in project 'MySite'. Do you want to overwrite it?" So make sure to only answer "**No**" (in the GUI) or "**N**" (in the package manager console).

![](images/nuget-overwrite-dialog.png)

> ![](images/nuget-upgrade-overwrite.png)

Reviewer: Looks like the images from the article on our has not been moved over.

Can't see them on github or gitbook, however, I do see them in the article on our:
- https://our.umbraco.com/documentation/Fundamentals/Setup/Upgrading/general#upgrades-to-versions-lower-than-720

Now here comes the tricky bit: We'll be mean and overwrite your web.config file anyway. But we'll back it up so don't worry. (Plus you already had your own backup, right?) You can find the backup in `App_Data\NuGetBackup\20140320-165450\`. (The `20140320-165450` bit is the date and time when the backup occurred, which varies.) You can then merge your config files and make sure they're completely up to date.

## Upgrade manually from a zip file

Download the .zip file for the new version you are upgrading to from [https://our.umbraco.com/download](https://our.umbraco.com/download)

Copy the following folders from inside the .zip file over the existing folders in your site:

* `/bin`
* `/Umbraco`
* `/Umbraco_Client`

{% hint style="info" %}

> There are hosting providers (we know of one: RackSpace Cloud) that require proper casing of file and folder names. Normally on Windows this is not a problem. If your hosting provider however forces proper casing, you will need to verify that the folder and file names are in the same casing as in the newest version you're upgrading to.

Reviewer: ```suggestion
There are hosting providers (we know of one: RackSpace Cloud) that require proper casing of file and folder names. Generally, on Windows, this is not a problem. Is your hosting provider forcing proper casing? You'll then need to verify that folders and files are named in the same casing as the version you're upgrading to.
```

{% endhint %}

## Merge configuration files

You can expect some changes to the following configuration files:

* Any file in the `/Config` folder
* The `/Global.asax` file
* The `web.config` file in the root of your site **(Important: make sure to copy back the version number, and the connection string as they were.)**
* In rare cases, the `web.config` file in the Views folder

Use a tool like [WinMerge](http://winmerge.org/ "WinMerge") to check changes between all of the config files. Depending on when you last did this there may have been updates to quite a few of them.

> There's also the possibility that some files in the `/Config` folder are new or some have been removed (we do make a note of this in the release notes). WinMerge (and other diff tools) is able to compare folders as well so you can spot these differences.

Reviewer: ```suggestion
There's also the possibility that files in the `/Config` folder are new or have been removed(we note this in the release notes). WinMerge (and other diff tools) is able to compare folders as well so you can spot these differences.
```

Packages like Contour, Umbraco Forms, and Courier also make changes to the language files located in: `/Umbraco/Config/Lang/*.xml` (typically `en.xml`).

## Finalize

After copying the files and making the config changes, you can open your site. You should see the installer which will guide you through the upgrade.

The installer will do two things:

* Update the version number in the `web.config`
* Upgrade your database in case there are any changes

We are aware that, currently, the installer is asking you for the database details of a **blank database** while upgrading. In the near future this will be pre-filled with your existing details and the wording will be updated. So no need to be scared. Enter the details of your existing database and Umbraco will upgrade it to the latest version when necessary.

## Post installation

One important recommendation is to always remove the `install` folder immediately after upgrading Umbraco and never to upload it to a live server.

## Potential issues and gotchas

### Browser cache

> Google Chrome has notoriously aggressive caching, so if something doesn't seem to work well in the backoffice, make sure to clear cache and cookies thoroughly (for other browsers as well). Normally the browser cache problem is automatically handled in an Umbraco upgrade by modifying the config/ClientDependency.config version number. If you however wish to re-force this update you can increment this version number which will ensure that any server-side cache of JavaScript and stylesheets gets cleared as well.

Reviewer: ```suggestion
Google Chrome has notoriously aggressive caching. If something doesn't seem to work well in the backoffice, make sure to clear cache and cookies thoroughly (for other browsers as well). Normally the browser cache problem is automatically handled in an Umbraco upgrade by modifying the config/ClientDependency.config version number. If you wish to re-force this update you can increment this version number. This will ensure that any server-side cache of JavaScript and stylesheets gets cleared as well.
```
